test/puppetlabs/puppetdb/reports_test.clj

```diff
@@ -50,3 +53,33 @@
   (testing "no action on valid reports"
     (let [test-data (:basic reports)]
       (= (sanitize-report test-data) test-data))))
+
+(defn underscore->dash-report-keys [m]
+  (->> m
+       utils/underscore->dash-keys
+       (sp/update [:resource-events sp/ALL sp/ALL]
```

Comment:
Same concern as above. Odds are good I simply don't understand the selector mechanism.